execute/table_test.go

```diff
@@ -253,3 +255,70 @@ func TestColListTable_SetNil(t *testing.T) {
 		t.Fatalf("unexpected error: %s", err)
 	}
 }
+
+func TestCopyTable(t *testing.T) {
+	alloc := &memory.Allocator{}
+
+	res, err := gen.Input(gen.Schema{
+		Tags: []gen.Tag{
+			{Name: "t0", Cardinality: 1},
+		},
+		NumPoints: 100,
+		Period:    time.Hour,
+		Types: map[flux.ColType]int{
+			flux.TFloat: 1,
+		},
+		Alloc: alloc,
+	})
+	if err != nil {
+		t.Fatalf("unable to generate tables: %s", err)
+	}
+
+	var buffers []flux.BufferedTable
+	for res.More() {
+		r := res.Next()
+		if err := r.Tables().Do(func(table flux.Table) error {
+			bt, err := execute.CopyTable(table)
+			if err != nil {
+				return err
+			}
+			buffers = append(buffers, bt)
+			return nil
+		}); err != nil {
+			t.Fatalf("unexpected error: %s", err)
+		}
+	}
+
+	// Ensure we can copy the table and read a point from the
+	// column reader without panicking.
+	for _, buf := range buffers {
+		cpy := buf.Copy()
```

Comment:
So would it be an error to call `Do(..)` on `buf` directly here without the call to `Copy()`?  If so, would be good to verify that...

Reply:
Yes it would. This is going to be part of the common table tests.